### PR TITLE
Load balancer

### PR DESCRIPTION
**Pull request by Alevsk**

### CODE REVIEW

File: k8s/load-balancer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM nginx:latest
```
Comment (dvaldivia) [inline k8s/load-balancer/Dockerfile]: I think we can use pure nginx without the need of our own image

**Comment by Alevsk** [comment]
@aead Hi Andreas, I'll like your opinion and ideas on this :), this PR build the nginx.config file being used by the nginx to resolve the tenants by querying the tenants table, right now its a dirty way to build the file but its not that bad since we are storing it on kubernetes configMap, on the other side I have some concern regarding user direct input is used to create the routing rules for nginx:

```
	server {
		server_name USER-CONTROL-THIS.s3.localhost;
		location / {
			proxy_pass http://USER-PARTIALLY-CONTROL-THIS:%d;
		}
	}
```

From the security perspective we should have a mechanism to avoid malicious to expose minio internal services by defining known service names (dashboard. kubedns, etc, etc)

File: cluster/storage-cluster.go [hunk 2]
```diff
@@ -187,6 +187,44 @@ func GetListOfTenantsForStorageGroup(ctx *Context, sg *StorageGroup) chan []*Sto
 	return ch
 }
 
+// GetListOfTenants returns a list of all tenants that currently exists on the cluster
```
Comment (dvaldivia) [inline cluster/storage-cluster.go]: update the comment

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — APPROVED** [review]

File: k8s/load-balancer/Dockerfile (hunk 1 above)
Comment (Alevsk) [reply k8s/load-balancer/Dockerfile]: that file is not need it, im removing the whole `load-balancer` folder since it was only for testing purposes.

**Pull request merged**